src/librustc_mir/interpret/cast.rs

```diff
@@ -108,7 +108,10 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                 // The src operand does not matter, just its type
                 match src.layout.ty.sty {
                     ty::Closure(def_id, substs) => {
-                        let substs = self.subst_and_normalize_erasing_regions(substs)?;
+                        let substs = self.subst_and_normalize_erasing_regions_in_frame(substs);
+                        if substs.needs_subst() {
+                            return err!(TooGeneric);
+                        }
```

Comment:
This seems like an awfully brittle API. So I first need to subst and then check if I need to subst more?

Reply:
see https://github.com/rust-lang/rust/pull/61041#discussion_r308014023